tools/build-templated-pages/src/features.rs

```diff
@@ -58,53 +38,88 @@ fn parse_features(panic_on_missing: bool) -> Vec<Feature> {
-                    .as_str()
-                    .unwrap()
-                    .to_string();
-                if let Some(description) = key.leaf_decor().prefix() {
-                    let description = description.as_str().unwrap().to_string();
-                    if !description.starts_with("\n# ") || !description.ends_with('\n') {
-                        panic!("Missing description for feature {name}");
+                let key_name = get_key_name(key);
+                let key_prefix = key.leaf_decor().prefix();
+                if what_to_run.contains(Command::CHECK_MISSING) {
+                    if let Some(description) = key_prefix {
+                        create_feature(&default, key_name, description)
+                    } else {
+                        panic!("Missing description for feature {key_name}");
                     }
-                    let description = description
-                        .strip_prefix("\n# ")
-                        .unwrap()
-                        .strip_suffix('\n')
-                        .unwrap()
-                        .to_string();
-                    Some(Feature {
-                        is_default: default.contains(&name),
-                        name,
-                        description,
-                    })
-                } else if panic_on_missing {
-                    panic!("Missing description for feature {name}");
                 } else {
-                    None
+                    let feature = |description| create_feature(&default, key_name, description);
+                    key_prefix.map(feature).unwrap()
                 }
             }
         })
         .collect()
 }
 
+fn get_manifest_features() -> Table {
+    let manifest_file = std::fs::read_to_string("Cargo.toml").unwrap();
+    let manifest = manifest_file.parse::<Document>().unwrap();
+    let features = manifest["features"].as_table().unwrap().clone();
+    features
+}
+
+fn create_feature(default: &[String], name: String, description: &RawString) -> Option<Feature> {
+    let description = description.as_str().unwrap().to_string();
+    if let Some(stripped_description) = description
+        .strip_prefix("\n# ")
+        .and_then(|d| d.strip_suffix('\n'))
+    {
+        let is_default = default.contains(&name);
+        let description = get_description(stripped_description.to_string(), &name);
+        Some(Feature {
+            is_default,
+            name,
+            description,
+        })
+    } else {
+        panic!("Missing description for feature {name}");
+    }
+}
+
+fn get_description(description: String, name: &String) -> String {
+    let description = description.to_string();
+    if !description.starts_with("\n# ") || !description.ends_with('\n') {
+        panic!("Missing description for feature {name}");
+    }
+    let stripped_description = description.strip_prefix("\n# ").unwrap().strip_suffix('\n');
+    stripped_description.unwrap().to_string()
+}
```

Comment:
```suggestion
fn get_description<'a>(description: &'a str, name: &str) -> &'a str {
    if !(description.starts_with("\n# ") && description.ends_with('\n')) {
        panic!("Missing description for feature {name}");
    }
    
    description
        .strip_prefix("\n# ")
        .and_then(|s| s.strip_suffix('\n'))
        .unwrap()
}
```

It is generally good practice to pass strings by reference, to prevent needless deallocations. You can also use `Option::and_then` to chain calls that return `Option`. You can also return a reference, since it's just a slice of `description`.

These changes will require modifications elsewhere in the code, but it's not too bad.